handler/src/main/java/io/netty/handler/ssl/OpenSslPrivateKey.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.internal.tcnative.SSL;
+import io.netty.util.AbstractReferenceCounted;
+import io.netty.util.IllegalReferenceCountException;
+
+import javax.security.auth.Destroyable;
+import java.security.PrivateKey;
+
+final class OpenSslPrivateKey extends AbstractReferenceCounted implements PrivateKey {
+
+    private volatile long privateKey;
+
+    OpenSslPrivateKey(long privateKey) {
+        this.privateKey = privateKey;
+    }
+
+    @Override
+    public String getAlgorithm() {
+        return null;
+    }
+
+    @Override
+    public String getFormat() {
+        return "native";
```

Comment:
https://github.com/netty/netty/pull/8200#discussion_r210673926 ... IIUC you meant to make this `null`?